nucypher/config/utils.py

```diff
@@ -134,3 +55,165 @@ def check_config_runtime() -> bool:
     return True
 
 
+def validate_nucypher_ini_config(config=None,
+                                 filepath: str=DEFAULT_INI_FILEPATH,
+                                 raise_on_failure: bool=False) -> Tuple[bool, list]:
+
+    if config is None:
+        config = configparser.ConfigParser()
+
+    try:
+        config.read(filepath)
+    except:
+        raise  # FIXME
+
+    required_sections = ("provider", "nucypher")
+
+    missing_sections = list()
+    for section in required_sections:
+        if section not in config.sections():
+            missing_sections.append(section)
+            if raise_on_failure is True:
+                raise RuntimeError("Invalid config file")
+    else:
+        if len(missing_sections) > 0:
+            return False, missing_sections
+
+
+def parse_blockchain_config(config=None, filepath: str=DEFAULT_INI_FILEPATH) -> dict:
+
+    if config is None:
+        config = configparser.ConfigParser()
+        config.read(filepath)
+
+    providers = list()
+    if config['provider']['type'] == 'ipc':
+        try:
+            provider = IPCProvider(config['provider']['ipc_path'])
+        except KeyError:
+            message = "ipc_path must be provided when using an IPC provider"
+            raise Exception(message)  # FIXME
+        else:
+            providers.append(provider)
+    else:
+        raise NotImplementedError
+
+    poa = config.getboolean(section='provider', option='poa', fallback=True)
```

Comment:
It feels to me that we're writing a schema with datatypes by doing the `getboolean` etc, which kinda suggests that we need some facility for that (hopefully, not XML lol). I think, could be not a bad idea to write the schema in Python.

But I'd leave this to the future: masering configurability of configs is not the focus of what we need right now. Can leave a little `# TODO` note though.

For now, this is great stuff